wcfsetup/install/files/lib/form/AccountManagementForm.class.php

```diff
@@ -338,12 +338,12 @@ public function save() {
 		
 		// email
 		if (WCF::getSession()->getPermission('user.profile.canChangeEmail') && $this->email != WCF::getUser()->email && $this->email != WCF::getUser()->newEmail) {
-			if (REGISTER_ACTIVATION_METHOD == 0 || REGISTER_ACTIVATION_METHOD == 2 || mb_strtolower($this->email) == mb_strtolower(WCF::getUser()->email)) {
+			if (!(REGISTER_ACTIVATION_METHOD & User::REGISTER_ACTIVATION_USER) && mb_strtolower($this->email) == mb_strtolower(WCF::getUser()->email)) {
```

Comment:
```suggestion
			if (!(REGISTER_ACTIVATION_METHOD & User::REGISTER_ACTIVATION_USER) || mb_strtolower($this->email) == mb_strtolower(WCF::getUser()->email)) {
```

```suggestion
			if (!(REGISTER_ACTIVATION_METHOD & User::REGISTER_ACTIVATION_USER)) {
```

Choose either of these. The second one is more correct from a theoretical perspective, because `TimWolla@example.com` is not guaranteed to be the same as `timwolla@example.com`.